### PR TITLE
Fix some MSVC warnings V2 + nihstro update

### PR DESCRIPTION
**Pull request by Kloen**
This PR could be bigger, but I think it's big enough.

Fixed about 30 cast warnings ([C4267](https://msdn.microsoft.com/en-us/library/6kck0s93.aspx) and [C4244](https://msdn.microsoft.com/en-us/library/th7a07tz.aspx))
```
C:\projects\citra\src\core\memory.cpp(395): warning C4267: 'initializing': conversion from 'size_t' to 'VAddr', possible loss of data 
C:\projects\citra\src\core\memory.cpp(395): warning C4267: 'initializing': conversion from 'size_t' to 'const VAddr', possible loss of data
C:\projects\citra\src\core\memory.cpp(418): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data 
C:\projects\citra\src\core\memory.cpp(426): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data 
C:\projects\citra\src\core\memory.cpp(465): warning C4267: 'initializing': conversion from 'size_t' to 'VAddr', possible loss of data 
C:\projects\citra\src\core\memory.cpp(465): warning C4267: 'initializing': conversion from 'size_t' to 'const VAddr', possible loss of data
C:\projects\citra\src\core\memory.cpp(489): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data 
C:\projects\citra\src\core\memory.cpp(498): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data 
C:\projects\citra\src\core\memory.cpp(523): warning C4267: 'initializing': conversion from 'size_t' to 'VAddr', possible loss of data 
C:\projects\citra\src\core\memory.cpp(523): warning C4267: 'initializing': conversion from 'size_t' to 'const VAddr', possible loss of data
C:\projects\citra\src\core\memory.cpp(546): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data 
C:\projects\citra\src\core\memory.cpp(555): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data 
C:\projects\citra\src\core\memory.cpp(577): warning C4267: 'initializing': conversion from 'size_t' to 'VAddr', possible loss of data 
C:\projects\citra\src\core\memory.cpp(577): warning C4267: 'initializing': conversion from 'size_t' to 'const VAddr', possible loss of data
C:\projects\citra\src\core\memory.cpp(601): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data 
C:\projects\citra\src\core\memory.cpp(609): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data 
C:\projects\citra\src\core\memory.cpp(622): warning C4267: '+=': conversion from 'size_t' to 'VAddr', possible loss of data
C:\projects\citra\src\core\memory.cpp(623): warning C4267: '+=': conversion from 'size_t' to 'VAddr', possible loss of data
C:\projects\citra\src\core\hle\kernel\thread.cpp(406): warning C4267: 'initializing': conversion from 'size_t' to 'u32', possible loss of data
C:\projects\citra\src\core\hle\kernel\thread.cpp(414): warning C4267: '=': conversion from 'size_t' to 'u32', possible loss of data
C:\projects\citra\src\core\frontend\emu_window.cpp(101): warning C4244: '=': conversion from 'float' to 's16', possible loss of data
C:\projects\citra\src\core\frontend\emu_window.cpp(102): warning C4244: '=': conversion from 'float' to 's16', possible loss of data
C:\projects\citra\src\core\frontend\emu_window.cpp(103): warning C4244: '=': conversion from 'float' to 's16', possible loss of data
C:\projects\citra\src\core\frontend\emu_window.cpp(112): warning C4244: '=': conversion from 'float' to 's16', possible loss of data
C:\projects\citra\src\core\frontend\emu_window.cpp(113): warning C4244: '=': conversion from 'float' to 's16', possible loss of data
C:\projects\citra\src\core\frontend\emu_window.cpp(114): warning C4244: '=': conversion from 'float' to 's16', possible loss of data
C:\projects\citra\src\core\hle\service\dsp_dsp.cpp(142): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
C:\projects\citra\src\core\hle\service\fs\archive.cpp(221): warning C4267: 'argument': conversion from 'size_t' to 'const u32', possible loss of data
C:\projects\citra\src\core\hle\applets\erreula.cpp(35): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data
C:\projects\citra\src\core\hle\applets\mii_selector.cpp(43): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data
C:\projects\citra\src\core\hle\applets\swkbd.cpp(46): warning C4267: 'argument': conversion from 'size_t' to 'u32', possible loss of data
C:\projects\citra\src\core\hle\kernel\shared_memory.cpp(38): warning C4267: '=': conversion from 'size_t' to 'u32', possible loss of data
````

Updated nihstro to latest master, to fix more warnings due to a upstream change I made that got merged recently (https://github.com/neobrain/nihstro/pull/54).
There are about 14 instances of this one:
```
C:\projects\citra\externals\nihstro\include\nihstro\shader_bytecode.h(764): warning C4800: 'unsigned int': forcing value to bool 'true' or 'false' (performance warning)
```

### CODE REVIEW

File: src/core/hle/service/dsp_dsp.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ static void LoadComponent(Service::Interface* self) {
     Memory::ReadBlock(buffer, component_data.data(), component_data.size());
 
     LOG_INFO(Service_DSP, "Firmware hash: %#" PRIx64,
-             Common::ComputeHash64(component_data.data(), component_data.size()));
+             Common::ComputeHash64(component_data.data(), static_cast<int>(component_data.size())));
```
Comment (lioncash) [inline src/core/hle/service/dsp_dsp.cpp]: The preferable way to resolve this one is to make the length parameter of ComputeHash64 take a size_t. Ditto for `RasterizerFlushRegion` and `RasterizerFlushAndInvalidateRegion`

**Comment by Kloen** [comment]
**Current master:** 93 warnings on MSVC
**Master + this:** 47 Warnings on MSVC

File: src/core/hle/service/dsp_dsp.cpp (hunk 1 above)
Comment (Kloen) [inline src/core/hle/service/dsp_dsp.cpp]: I was waiting on someone to suggest that change in fact. Didn't want to go all th way with it by myself.

My proposal:

For ComputeHash64: https://gist.github.com/Kloen/f0131281f1db93695e056de3a344a8c4
For RasterizeFlushRegion and RasterizerFlushAndInvalidateRegion: https://gist.github.com/Kloen/fc5997e3582cb30e2b7340c6317694e4

Comment (yuriks) [inline src/core/hle/service/dsp_dsp.cpp]: The Rasterizer sizes are limited to the 32-bit address space of the 3DS, so I don't think they should take `size_t`. (Though maybe there's a good reason for them to?)

Comment (wwylele) [inline src/core/hle/service/dsp_dsp.cpp]: @Kloen for ComputeHash64 I think you can put your proposal here. For RasterizeFlushRegion, I kind of like keeping the parameter as u32 but not sure. Probably you may want to move it to another PR so we can discuss this.

**Comment by freiro** [comment]
Seems legit, I would leave additional fixes to a different PR.

**Comment by Kloen** [comment]
I'll bisect this PR into smaller PR's with the major changes on them in order to merge it by pieces while we can discuss the important topics separately.
